### ISSUE TITLE
Enhancement suggestion: Ignored base templates should not be removed from template.TemplateInfo.BaseTemplateIds

### ISSUE THREAD

**Issue by vhil** [blipson89/Leprechaun#33]
At the moment, if I use default config which has the template exclusion rules on `<templateReader>` node: 
`<excludedBaseTemplate id="{8CA06D6A-B353-44E8-BC31-B528C7306971}" name="Rendering Parameters Template" />`
I will have no possibility to know if the template is inherited from rendering parameters template or not.

What I would expect from this type of exclusion is that if we ignore that base template, it is obviously not included into de-serialized base templates property, as well as no fields from that base are template included into `AllFields` propery. However, I would still like to see base template ID in the `template.TemplateInfo.BaseTemplateIds` property. 

That would mean that Leprechaun did read the .yaml properly, however did not de-serialize ignored base template.

Would that make sense?

**Comment by blipson89** [comment]
Yep! I'll add this to the backlog. I have a few ideas, but I need to figure out which one will work best.
